vertx/src/main/java/com/microsoft/gctoolkit/vertx/DataSourceVerticle.java

```diff
@@ -0,0 +1,68 @@
+package com.microsoft.gctoolkit.vertx;
+
+import com.microsoft.gctoolkit.io.GCLogFile;
+import com.microsoft.gctoolkit.message.DataSourceChannelListener;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class DataSourceVerticle extends AbstractVerticle {
+
+    private static final Logger LOGGER = Logger.getLogger(DataSourceVerticle.class.getName());
+
+    final private Vertx vertx;
+    final private String inbox;
+    private String id;
+    final private DataSourceChannelListener processor;
+    public DataSourceVerticle(Vertx vertx, String channelName, DataSourceChannelListener listener) {
+        this.vertx = vertx;
+        this.inbox = channelName;
+        this.processor = listener;
+    }
+
+    public void setID(String id) {
+        this.id = id;
+    }
+
+    @Override
+    public void start(Promise<Void> promise) {
+        try {
+            vertx.eventBus().<String>consumer(inbox, message -> {
+                processor.receive(message.body());
+                if (GCLogFile.END_OF_DATA_SENTINEL.equals(message.body())) {
```

Comment:
Should this be in stop(Promise) ?

Reply:
This callback is what sees the EOF sentinel. The close is called when vertx is closed.